using --semantic-release to determine next version is unreliable

### Summary 

Semantic release fails with latest JX.

```
Showing logs for build garethjevans-test-semantic-version-golang-http-master-3 stage meta-pipeline and container step-create-tekton-crds
running command: jx step next-version --semantic-release --tag
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c749f1]
goroutine 1 [running]:
github.com/Masterminds/semver.(*Version).String(0x0, 0x0, 0xc00087a120)
	/workspace/go/pkg/mod/github.com/!masterminds/semver@v1.4.2/version.go:113 +0x51
github.com/jenkins-x/jx/v2/pkg/cmd/step.(*StepNextVersionOptions).Run(0xc0005d6a80, 0x0, 0x0)
	/workspace/source/pkg/cmd/step/step_next_version.go:125 +0x238
github.com/jenkins-x/jx/v2/pkg/cmd/step.NewCmdStepNextVersion.func1(0xc000770a00, 0xc00081d8e0, 0x0, 0x2)
	/workspace/source/pkg/cmd/step/step_next_version.go:91 +0x71
github.com/spf13/cobra.(*Command).execute(0xc000770a00, 0xc00081d8c0, 0x2, 0x2, 0xc000770a00, 0xc00081d8c0)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000320f00, 0xc00058f640, 0x51f7dc0, 0xc0000d0000)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/jenkins-x/**/v2/cmd/**/app.Run(0x0, 0x0, 0x0, 0x3e8ea40, 0xc0000b4058)
	/workspace/source/cmd/jx/app/jx.go:19 +0x159
main.main()
	/workspace/source/cmd/jx/jx.go:11 +0x32
error: failed to set the version on release pipelines: failed to run '/bin/sh -c jx step next-version --semantic-release --tag' command in directory '/workspace/source', output: ''
Pipeline failed on stage 'meta-pipeline' : container 'step-create-tekton-crds'. The execution of the pipeline has stopped.
```

Pipeline is configured with 

```
buildPack: go
pipelineConfig:
  pipelines:
    release:
      setVersion:
        replace: true
        steps:
          - name: next-version
            sh: jx step next-version --semantic-release --tag
            comment: set the version number using semantic-release and create a new release tag
```

--- a/pkg/cmd/step/step_next_version.go
+++ b/pkg/cmd/step/step_next_version.go
@@ -114,6 +114,7 @@ func (o *StepNextVersionOptions) Run() error {
 		if err != nil {
 			return errors.WithStack(err)
 		}
+		log.Logger().Infof("latest tag %s and rev %s", util.ColorInfo(tag), util.ColorInfo(rev))
 		cur, err := o.Git().RevParse(o.Dir, "HEAD")
 		if err != nil {
 			return errors.WithStack(err)

--- a/pkg/semrel/semrel.go
+++ b/pkg/semrel/semrel.go
@@ -6,6 +6,8 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/jenkins-x/jx-logging/pkg/log"
+
 	"github.com/pkg/errors"
 
 	"github.com/Masterminds/semver"
@@ -48,11 +50,13 @@ func calculateChange(commits []*conventionalCommit, latestRelease *release) chan
 }
 
 func applyChange(version *semver.Version, change change) *semver.Version {
+	log.Logger().Debugf("applying change %+v", change)
 	if version.Major() == 0 {
 		change.Major = true
 	}
 	if !change.Major && !change.Minor && !change.Patch {
-		return nil
+		log.Logger().Debugf("unable to determine change so defaulting to patch")
+		change.Patch = true
 	}
 	var newVersion semver.Version
 	preRel := version.Prerelease()
@@ -102,6 +106,7 @@ func GetNewVersion(dir string, endSha string, gitter gits.Gitter, latestTag stri
 		return nil, errors.Wrapf(err, "getting commits in range %s..%s", release.SHA, endSha)
 	}
 	commits := make([]*conventionalCommit, 0)
+	log.Logger().Debugf("got %d commits", len(rawCommits))
 	for _, c := range rawCommits {
 		commit := c
 		commits = append(commits, parseCommit(&commit))
@@ -114,6 +119,7 @@ func parseCommit(commit *gits.GitCommit) *conventionalCommit {
 	c := &conventionalCommit{
 		GitCommit: commit,
 	}
+	log.Logger().Debugf("parsing message '%s'", commit.Message)
 	c.MessageLines = strings.Split(commit.Message, "\n")
 	found := commitPattern.FindAllStringSubmatch(c.MessageLines[0], -1)
 	if len(found) < 1 {

